addons/project/models/project_project.py

```diff
@@ -208,11 +208,34 @@ def _read_group_stage_ids(self, stages, domain, order):
     milestone_count = fields.Integer(compute='_compute_milestone_count', groups='project.group_project_milestone')
     milestone_count_reached = fields.Integer(compute='_compute_milestone_reached_count', groups='project.group_project_milestone')
     is_milestone_exceeded = fields.Boolean(compute="_compute_is_milestone_exceeded", search='_search_is_milestone_exceeded')
+    milestone_progress = fields.Integer("Milestones Reached", compute='_compute_milestone_reached_count')
+    next_milestone_id = fields.Many2one('project.milestone', compute='_compute_next_milestone_id', store=True)
+    is_next_milesotone_unreached_and_late = fields.Boolean(compute='_compute_is_next_milesotone_unreached_and_late')
 
     _sql_constraints = [
         ('project_date_greater', 'check(date >= date_start)', "The project's start date must be before its end date.")
     ]
 
+    @api.depends('milestone_ids', 'milestone_ids.is_reached', 'milestone_ids.deadline')
+    def _compute_next_milestone_id(self):
+        projects_read_group = {project.id: milestone_list for project, milestone_list in self.env['project.milestone']._read_group(
+            [('project_id', 'in', self.ids), ('is_reached', '=', False)],
+            ['project_id'],
+            ['id:array_agg'],
+        )}
+        for project in self:
+            project.next_milestone_id = projects_read_group.get(project.id, False) and projects_read_group[project.id][0]
```

Comment:
```suggestion
        milestone_ids_per_project_id = {
            project.id: milestone_ids
            for project, milestone_ids in self.env['project.milestone']._read_group(
                [('project_id', 'in', self.ids), ('is_reached', '=', False)],
                ['project_id'],
                ['id:array_agg'],
            )}
        }
        for project in self:
            project.next_milestone_id = project.id in milestone_ids_per_project_id and milestone_ids_per_project_id[project.id][0]
```
Glad you found a solution :smiley: Would just improve it a little bit:
- Name the dict `<key>_per_<value>`
- Some indents
- When you only want to check if in dict, use `key in dict` rather than `dict.get(key)`

Reply:
Done!